src/bin/wd_copier.rs

```diff
@@ -15,7 +18,11 @@ struct Opt {
 fn main() {
     let opt = Opt::from_args();
 
+    let meminfo = procfs::Meminfo::new().unwrap();
+    let starting_dirty = meminfo.dirty;
```

Comment:
Since the meminfo stuff is not critical to functionality, I'd rather it not use unwrap. Maybe add a new function that just returns 0 if `Meminfo::new` fails:

```rust
fn get_dirty_bytes() -> u64 { ... }
```

(I'm not sure if the return value is in bytes, so maybe this function name is wrong)

Reply:
I think that makes sense, and it makes the scopes much clearer than just having a bunch of Object::new constructors called inline in main.

Reply:
Hmm, shouldn't the `starting` value be calculated after the copy operation? Otherwise what is it measuring?

Reply:
It calculates based on the amount of dirty bytes before and after the copy, using that as a baseline.

My first implementation didn't have the earlier `starting_dirty` and seemed to work OK, but I'd think this might be more accurate on a currently active system.

Reply:
Ah right, that makes sense

Reply:
dbc2118